### ISSUE TITLE
Can't reconnect with server after a network shutdown.

### ISSUE THREAD

**Issue by SergioBertolinSG** [owncloud/client#2321]
### Steps to reproduce
1. Have a well configured owncloud desktop session
2. Disconnect network
3. Wait for the owncloud client error
4. Reconnect network

### Expected behaviour
Client reconnects to the server after some time

### Actual behaviour
Client keeps the error and can't reconnect, even clicking on sign in.

### Server configuration
**Operating system**:
Linux/SUSE

**Web server:**
Apache/2.2.17

**Are you using encryption:** no

### Client configuration

**Operating system:**

Windows 7

Tested in Mac OS 1.9.5, it doesn't happen there.

#### Client logs

https://gist.github.com/SergioBertolinSG/b521b4d6cad127577424

**Comment by guruz** [comment]
Could be fixed in 1.8 with the different network library

**Comment by dragotin** [comment]
This is **not** windows only, I can reproduce it easily: Just switch off the Wifi on the notebook, the client wont reconnect once the Wifi is back again.

**Comment by guruz** [comment]
@dragotin But it properly disconnects? What did you see in the log? If I remember correctly the CheckServerJob should be scheduled and then see that it works again
(Although I believe the scheduling of those jobs might be messed up https://github.com/owncloud/client/issues/2801 )

**Comment by guruz** [comment]
From @SergioBertolinSG 's log:

```
10-16 13:38:05:337 networkjobs.cpp:427 status.php returns:  QMap()   99  Reply:  QNetworkReplyHttpImpl(0x7628da8)
10-16 13:38:05:337 networkjobs.cpp:433 No proper answer on   QUrl( "https://beta.owncloud.com/owncloud/status.php" ) 
10-16 13:38:27:345 networkjobs.cpp:201 !!! Mirall::CheckServerJob created for  QUrl( "https://beta.owncloud.com/owncloud" )  querying "status.php"
10-16 13:38:37:343 networkjobs.cpp:146 void Mirall::AbstractNetworkJob::slotFinished() 99 "Connection timed out"
```

But @SergioBertolinSG you said you reconnected the WiFi, right? Even inside the time frame of your log?
(Internal note: Might be similar to https://github.com/owncloud/client/issues/1312#issuecomment-53903942 .. in worst case we need to recreate the QNetworkAccessManager)

**Comment by ckamm** [comment]
While I could not reproduce it (maybe I didn't wait long enough in the disconnected state), I did spot a bad account state transition. If the network is unavailable, the state goes from Connected -> Timeout. But then, because we can't reach the server's status.php, it goes to ConfigurationError.

That should not happen, will fix.

**Comment by ckamm** [comment]
@lmaestro I now see the regular

```
networkjobs.cpp:228 !!! OCC::CheckServerJob created for ...  querying "status.php" 
networkjobs.cpp:162 void OCC::AbstractNetworkJob::slotFinished() 3 "Host  not found" 
```

when I disconnect my network.

**Comment by lmaestro** [comment]
Fixed in version 1.8.0-nightly20150212 (build 4714)
